core/src/main/kotlin/org/evomaster/core/problem/rest/service/AbstractRestSampler.kt

```diff
@@ -80,13 +84,43 @@ abstract class AbstractRestSampler : HttpWsSampler<RestIndividual>() {
         val openApiSchema = problem.openApiSchema
 
         if(!openApiURL.isNullOrBlank()) {
-            swagger = OpenApiAccess.getOpenAPIFromURL(openApiURL)
+
+            // first try to retrieve the OpenAPI without authentication
+            swagger = OpenApiAccess.getOpenAPIFromURL(openApiURL, HttpWsNoAuth())
+
+            // if it fails, try to get swagger using each authentication information.
+            // if swagger failed due to unauthorized access, then an empty swagger is retrieved
+            if(swagger.paths == null) {
+
+                var currentAuthenticationIndex = 0
+                val authenticationList:List<AuthenticationDto> = infoDto.infoForAuthentication
```

Comment:
this will not work for black-box testing.
rather use `HttpWsSampler.authentications`

Reply:
to add to this, should have a E2E for black-box testing, where driver (useful for writing E2E but not needed for BB) has no auth info. auth can then be provided for `endpointCallLogin` via a yaml/toml configuration file. see for example how done in `LoginTokenEMTest`

Reply:
however, might need to move up `setupAuthentication(infoDto)` after reading `infoDto`, otherwise `authentications` will be empty here

Reply:
need to handle this also inside `fun initForBlackBox() `